designer/server/src/test/scala/pl/touk/nussknacker/ui/api/ComponentResourcesApiSpec.scala

```diff
@@ -0,0 +1,314 @@
+        val componentId = defaultComponentIdProvider.createComponentId(
+          TestProcessingTypes.Streaming,
+          Some(sourceComponentName),
+          ComponentType.Source
+        )
+
+        given()
+          .auth()
+          .basic("admin", "admin")
+          .and()
+          .pathParam("componentId", componentId.value)
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/components/{componentId}/usages")
+          .Then()
+          .statusCode(200)
+          .body(
+            matchJsonWithRegexValues(
+              s"""[{
+                 |  "id": "${processName.value}",
+                 |  "name": "${processName.value}",
+                 |  "processId": ${processId.value},
+                 |  "nodesUsagesData": [ { "nodeId": "source", "type": "ScenarioUsageData" } ],
+                 |  "isFragment": false,
+                 |  "processCategory": "$Category1",
+                 |  "modificationDate": "^\\\\d{4}-\\\\d{2}-\\\\d{2}T\\\\d{2}:\\\\d{2}:\\\\d{2}.\\\\d{6}Z$$",
+                 |  "modifiedAt": "^\\\\d{4}-\\\\d{2}-\\\\d{2}T\\\\d{2}:\\\\d{2}:\\\\d{2}.\\\\d{6}Z$$",
+                 |  "modifiedBy": "admin",
+                 |  "createdAt": "^\\\\d{4}-\\\\d{2}-\\\\d{2}T\\\\d{2}:\\\\d{2}:\\\\d{2}.\\\\d{6}Z$$",
+                 |  "createdBy": "admin",
+                 |  "lastAction": null
+                 |}]""".stripMargin
+            )
+          )
+      }
+
+      "return 404 when component not exist" in {
+        val badComponent: ComponentId = ComponentId("not-exist-component")
+
+        given()
+          .pathParam("componentId", badComponent.value)
+          .and()
+          .auth()
+          .basic("admin", "admin")
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/components/{componentId}/usages")
+          .Then()
+          .statusCode(404)
+          .body(equalTo(s"Component ${badComponent.value} not exist."))
+      }
+      "return 405 when invalid HTTP method is passed" in {
+        given()
+          .auth()
+          .basic("admin", "admin")
+          .when()
+          .put(s"$nuDesignerHttpAddress/api/components/id/usages")
+          .Then()
+          .statusCode(405)
+          .body(
+            equalTo(
+              s"HTTP method not allowed, supported methods: GET"
+            )
+          )
+      }
+    }
+
+    "not authenticated should" - {
+      "forbid access" in {
+        given()
+          .auth()
+          .none()
+          .when()
+          .get(s"$nuDesignerHttpAddress/api/components/id/usages")
+          .Then()
+          .statusCode(401)
+          .body(
+            equalTo(
+              "The resource requires authentication, which was not supplied with the request"
+            )
+          )
+      }
+    }
+  }
+
+  implicit class ExtractColumn[T <: ValidatableResponse](validatableResponse: T) {
+
+    def extractToList(column: String): List[String] = {
```

Comment:
1. it's rather `rootKey`, right?
2. and maybe it'd be better if we call it `extractToStringsList` (it's more true ;))

Reply:
To be sure, `rootKey` as instead of `column`?